server/src/main/java/io/crate/expression/operator/any/AnyNotLikeOperator.java

```diff
@@ -83,7 +87,7 @@ protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, Literal
     @Override
     protected Query literalMatchesAnyArrayRef(Function any, Literal<?> probe, Reference candidates, Context context) {
         String pattern = (String) probe.value();
-        String regexString = LikeOperators.patternToRegex(pattern, LikeOperators.DEFAULT_ESCAPE, false);
+        String regexString = LikeOperators.patternToRegex(pattern, LikeOperators.DEFAULT_ESCAPE);
```

Comment:
I think it's bug (should be true on master) but:

1. No change entry because this fix is mentioned in unreleased https://github.com/crate/crate/pull/15149, basically follow up. Linked PR didn't cover this because it doesn't hit Lucene query on array reference code path.

2. This fix is anyway not enough because
we allowed pattern be on the either side of LIKE ANY in https://github.com/crate/crate/commit/97ca0828aa3a52b7a7f64e98f57686a857ccd7aa but only for non-lucine-query execution (only in evaluate).


https://github.com/crate/crate/commit/97ca0828aa3a52b7a7f64e98f57686a857ccd7aa has to be reflected in optimized execution as well, when it's possible to build a Lucene query but this is unrelated to ESCAPE, I will follow up on this.